### PR TITLE
Fix AIX serial number collection

### PR DESCRIPTION
**Pull request by dlesel**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Hi @dlesel 
Are you sure your change is not breaking any AIX version out in the wild ? 
If yes, then you'll need to re-run the test-data for aix to make Travis happy.

File: includes/polling/os/unix.inc.php [hunk 1]
```diff
@@ -165,9 +165,12 @@
     $aix_descr = explode("\n", $device['sysDescr']);
     # AIX standard snmp deamon
     if ($aix_descr[1]) {
-        $serial = explode("Processor id: ", $aix_descr[1])[1];
+        $serial = substr(snmp_get($device, 'aixSeSerialNumber.0', '-Oqv', 'IBM-AIX-MIB'), -7);
```
Comment (PipoCanaja) [inline includes/polling/os/unix.inc.php]: Could you add a comment here with a sample ```aixSeSerialNumber``` to make it clear why the '-7' is needed ?

Comment (dlesel) [reply includes/polling/os/unix.inc.php]: > Hi @dlesel
> Are you sure your change is not breaking any AIX version out in the wild ?
> If yes, then you'll need to re-run the test-data for aix to make Travis happy.

i am improving. TL is part of AIX Verison: https://www.ibm.com/developerworks/community/blogs/AIXDownUnder/entry/understanding_aix_levels6?lang=en

**Comment by dlesel** [comment]
> Hi @dlesel
> Are you sure your change is not breaking any AIX version out in the wild ?
> If yes, then you'll need to re-run the test-data for aix to make Travis happy.

**Comment by PipoCanaja** [comment]
@dlesel Still not sure that you are not breaking things for existing users. We have to be convinced before we merge this, to avoid breaking things for existing setups.

**Comment by dlesel** [comment]
Probably this PR brakes errors in initial AIX definitions #10569. In #10569  Serial Number is incorrect and missing Machine hardware type. This PR fixes both of them.  I have added TL (Technology Level) to AIX version. This is very important because you have to submit TL when open case with IBM and some big clinical systems (for example Cerner) require specific TL level. This PR is about fixing bugs  in #10569 which generates incorrect or incomplete information.

**Comment by Jellyfrog** [comment]
Please check broken test; https://travis-ci.com/librenms/librenms/jobs/258705796

**Comment by PipoCanaja** [comment]
@fbourqui Could you have a look at this PR and let us know how it behave with your devices ? In term of Serial Numbers, versions, etc? 
Thanx

**Comment by PipoCanaja** [comment]
@dlesel Could you have a look at the tests and fix them please ?

**Comment by dlesel** [comment]
Travis uses AIX 5.3 which is out of support from 2012-04-30. AIX 5.3 has issues with SNMP. I have implemented this PR in production in September 2019. LibreNMS monitors about 40 AIX servers starting with version 6.1, 7.1 and 7.2.

**Comment by PipoCanaja** [comment]
@dlesel 
Travis uses our test-data database, which includes old and new devices. Mainly because people are using older devices in production, even if this should not be the case from a theoretical point of view ...
That being said, we need Travis to pass to be able to merge your change. So if your change is "safe" (meaning if you are 100% sure you don't break support for older devices) then you can re-process the existing test data (using `./scripts/save-test-data.php -o <os> -v <variant>`) and commit the resulting files. That'll make Travis happy, and make this PR ready for merge. 

We are also expecting @fbourqui to test as well, as he seems to have some devices.

File: includes/polling/os/unix.inc.php [hunk 2]
```diff
@@ -165,9 +165,14 @@
     $aix_descr = explode("\n", $device['sysDescr']);
     # AIX standard snmp deamon
     if ($aix_descr[1]) {
-        $serial = explode("Processor id: ", $aix_descr[1])[1];
+        # IBM Power Serial Number has 7 characters. We need last 7 characters from aixSeSerialNumber.0:
+        # aixSeSerialNumber.0 = "IBM,02061F57P"
+        $serial = substr(snmp_get($device, 'aixSeSerialNumber.0', '-Oqv', 'IBM-AIX-MIB'), -7);
```
Comment (murrant) [inline includes/polling/os/unix.inc.php]: Assuming the serial number will always have 7 characters is a good way to run into issues.  Better to split the output or use a regex.

Comment (dlesel) [reply includes/polling/os/unix.inc.php]: Hard to believe that IBM will change their own standard. Serial Number is deeply embedded into hardware. But you made a good point. I'll change the code and split by comma.

Comment (dlesel) [reply includes/polling/os/unix.inc.php:170]: i am under order to work from home all this month. my LibreNMS development server does not work with company's VPN. So i can not submit changes to this PR in April.

Comment (murrant) [reply includes/polling/os/unix.inc.php:170]: You can pull down this branch on any machine and make changes.

**Comment by Jellyfrog** [comment]
Closing due to inactivity, feel free to reopen

**Pull request closed without merging**